### PR TITLE
Geolocation

### PR DESCRIPTION
**Pull request by danni**
Geolocation. Depends on #74 (merges that branch a couple of times).

So far this:
- wraps Google Maps JS API in promises
- wraps geolocation in a promise
- obtains your geolocation and reverse geocodes it
- updates the UI with your reverse geocode

Notes:
- What is correct mechanism for storing Google Maps API keys when code is in Github? Currently uses the env.
- The "Skip This" item is left out because I don't think you can do a meaningful search without some kind of location. Unless we attempt to use their GeoIP instead?

### CODE REVIEW

File: src/pages/CategoryPage.js [hunk 1]
```diff
@@ -53,27 +69,57 @@ class CategoryPage extends React.Component {
...
+                            onTouchTap={this.goBack.bind(this)}
+                        >
+                            <NavigationArrowBack />
+                        </mui.IconButton>
+                    }
+                />
+
+                <div>
+                    Searching for <b>{this.category.name}</b> in
+                    <b> {location} </b>
```
Comment (michaeladams) [inline src/pages/CategoryPage.js:108]: Would `<strong>` make more sense then `<b>`?
Edit:  damn markdown

File: src/pages/LocationPage.js [hunk 2]
```diff
@@ -2,27 +2,209 @@
...
+        });
     }
 
     render(): React.Element {
         return (
-            <div>
-                <mui.AppBar title="Personalise" />
+            <div className="LocationPage">
+                <mui.AppBar
+                    title="Personalise"
```
Comment (michaeladams) [inline src/pages/LocationPage.js:131]: Again - not sure if this is the HTML attr "title" - but would a screen reader pick this up?

File: src/server/HtmlDocument.js [hunk 3]
```diff
@@ -160,6 +164,11 @@ class HtmlDocument extends React.Component {
       <script dangerouslySetInnerHTML={{__html: issconfig}} />
       { script.map((src, k) => <script key={k} src={src} />) }
 
+      <script
+        src={`//maps.googleapis.com/maps/api/js?key=${GOOGLE_KEY}`}>
+      </script>
+
```
Comment (michaeladams) [inline src/server/HtmlDocument.js:170]: extra line?

File: src/pages/CategoryPage.js (hunk 1 above)
Comment (danni) [inline src/pages/CategoryPage.js:108]: This is just a placeholder for the moment. The styling is in a separate branch.

File: src/pages/LocationPage.js (hunk 2 above)
Comment (danni) [inline src/pages/LocationPage.js:131]: It is not. It is literally the title of the AppBar component.

File: src/server/HtmlDocument.js (hunk 3 above)
Comment (danni) [inline src/server/HtmlDocument.js:170]: jscs insisted on it.

**Pull request merged**